**Breast Cancer Detection Using the Kernel SVM Classifier**

Data Set Characteristics: Multivariate,
Number of Instances:699  

Attribute Information:

1. Sample code number: id number
2. Clump Thickness: 1 - 10
3. Uniformity of Cell Size: 1 - 10
4. Uniformity of Cell Shape: 1 - 10
5. Marginal Adhesion: 1 - 10
6. Single Epithelial Cell Size: 1 - 10
7. Bare Nuclei: 1 - 10
8. Bland Chromatin: 1 - 10
9. Normal Nucleoli: 1 - 10
10. Mitoses: 1 - 10
11. Class: (2 for benign, 4 for malignant)


Importing the Libraries



In [1]:
import pandas as pd

Importing the Data Set i.e The Breast Cancer Data Set

In [ ]:
dataset = pd.read_csv('breast_cancer.csv')
X = dataset.iloc[:, 1:-1].values
y = dataset.iloc[:, -1].values
print(X, y)

Splitting the Data Set

In [ ]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2)
print(X_train, X_test, y_train, y_test)

Training your Kernel SVM Classifier

In [22]:
from sklearn.svm import SVC
SVCclassifier = SVC(kernel = 'rbf')
SVCclassifier.fit(X_train, y_train) 

SVC(C=1.0, break_ties=False, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='scale', kernel='rbf',
    max_iter=-1, probability=False, random_state=None, shrinking=True,
    tol=0.001, verbose=False)

Testing your classifier against the Test Data

In [25]:
y_pred = SVCclassifier.predict(X_test)

Confusion Matrix and Accuracy Score to compare your predicted result and the true result

In [26]:
from sklearn.metrics import confusion_matrix, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(accuracy_score(y_test, y_pred))

[[92  3]
 [ 1 41]]
0.9708029197080292


We have achieved an accuracy of 97.08% with the Kernel SVM Classifier

Computing the accuracy using the K-Fold Cross Validator

In [27]:
from sklearn.model_selection import cross_val_score
accuracies = cross_val_score(estimator = KNNclassifier, X = X_train, y = y_train, cv = 10)
print("Accuracy: {:.2f} %".format(accuracies.mean()*100))
print("Standard Deviation: {:.2f} %".format(accuracies.std()*100))

Accuracy: 97.07 %
Standard Deviation: 1.47 %
